homeassistant/components/mobile_app/entity.py

```diff
@@ -33,7 +35,9 @@ def unique_id(webhook_id, sensor_unique_id):
 class MobileAppEntity(RestoreEntity):
     """Representation of an mobile app entity."""
 
-    def __init__(self, config: dict, device: DeviceEntry, entry: ConfigEntry) -> None:
+    def __init__(
+        self, config: dict, device: DeviceEntry | str | None, entry: ConfigEntry
```

Comment:
`DeviceEntry | str` ? that seems like a bug.

Reply:
> `DeviceEntry | str` ? that seems like a bug.

It would only be a bug if it were used :)
I removed it: https://github.com/home-assistant/core/pull/67091/commits/0aab4903a95deb2578791fe4e593d38737c2ebe5